c_emulator/rvfi_dii.h

```diff
@@ -9,8 +254,6 @@ enum rvfi_prestep_t {
   RVFI_prestep_ok,        // Ready for step
 };
 
-typedef void (*packet_reader_fn)(lbits *rop, unit);
-
 class rvfi_handler {
```

Comment:
```suggestion
class rvfi_handler : public callbacks_if {
```

I think it would be simpler to implement the callback interface here, then you don't need all the delegation functions in riscv_callbacks_rvfi.